src/Splitio-net-core/Services/SegmentFetcher/Classes/SegmentTaskWorker.cs

```diff
@@ -41,11 +42,11 @@ public void ExecuteTasks(CancellationToken token)
                     //Wait indefinitely until a segment is queued
                     if (SegmentTaskQueue.segmentsQueue.TryTake(out segment, -1))
                     {
-                        Log.Info(string.Format("Segment dequeued: {0}", segment.name));
+                        Log.Debug(string.Format("Segment dequeued: {0}", segment.name));
```

Comment:
is there in .Net something like:

``` 
if(Log.isDebugEnabled()) {
Log.Debug(string.Format("Segment dequeued: {0}", segment.name)) 
}```

Why?

Usually, at least in java, the VM contractus all the objects to pass as params to the method, then enters the method to find you are NOT in debug mode, so nothing gets logged.

Reply:
Yes there is. The thing is that building the log message is expensive only if the ToString of an object that we are logging is expensive (which is not the case for the entries we are logging). 
Anyway, if we invoke `Log.Debug` only if `Log.isDebugEnabled()` we are doing no harm so I will include it